src/pages/accounts/ForgotPassword.tsx

```diff
@@ -81,37 +128,177 @@ function ForgotPasswordPage() {
-						if (validEmail(e.target.value)) setEmailErr(false);
-						setEmail(e.target.value);
+						if (
+							validEmail(e.target.value) ||
+							e.target.value === ''
+						) {
+							setEmail({ error: false });
+						}
+						setEmail({
+							value: e.target.value,
+						});
 					}}
-					onBlur={(e) => {
-						if (!validEmail(e.target.value)) setEmailErr(true);
+					onBlur={() => {
+						if (!validEmail(email.value) && email.value !== '') {
+							email.errorMessage = EMAIL_ERROR_MESSAGE;
+							setEmail({ error: true });
+						}
 					}}
-					label="Email"
-					required
-					error={emailErr}
-					placeholder="Email"
 				/>
-				<SubmitButton
-					variant="primary"
-					disabled={emailErr}
-					onClick={handleSubmit}
-				>
+
+				{codeStep && (
+					<TextInput
+						value={validationCode.value}
+						required
+						label="Validation Code"
+						error={validationCode.error}
+						placeholder="Validation Code"
+						errorMessage={validationCode.errorMessage}
+						disabled={validationCode.valueSet}
+						onChange={(e) => {
+							setValidationCode({ value: e.target.value });
+						}}
+						onBlur={() => {
+							if (
+								!validCode(validationCode.value) &&
+								validationCode.value !== ''
+							) {
+								validationCode.errorMessage = VALIDATION_CODE_ERROR_MESSAGE;
+								setValidationCode({ error: true });
+							}
+						}}
+					/>
+				)}
+				{passStep && (
+					<>
+						<TextInput
+							value={password.value}
+							label="Set New Password"
+							required
+							type="password"
+							error={password.error}
+							errorMessage={password.errorMessage}
+							onChange={(e) => {
+								if (
+									validPass(e.target.value) ||
+									e.target.value === ''
+								) {
+									setPassword({ error: false });
+								}
+								setPassword({ value: e.target.value });
+							}}
+							onBlur={() => {
+								if (
+									!validPass(password.value) &&
+									password.value !== ''
+								) {
+									password.errorMessage = PASS_ERROR_MESSAGE;
+									setPassword({ error: true });
+								}
+							}}
+						/>
+						<TextInput
+							value={confirmPass.value}
+							label="Confirm New Password"
+							required
+							type="password"
+							error={confirmPass.error}
+							errorMessage={confirmPass.errorMessage}
+							onChange={(e) => {
+								if (password.value === e.target.value) {
+									setConfirmPass({ error: false });
+								}
+								setConfirmPass({ value: e.target.value });
+							}}
+							onBlur={() => {
+								if (password.value !== confirmPass.value) {
+									if (confirmPass.value !== '') {
+										confirmPass.errorMessage = CONFIRM_PASS_ERROR_MESSAGE;
+										setConfirmPass({ error: true });
+									} else {
+										confirmPass.errorMessage = '';
+										setConfirmPass({ error: false });
+									}
+								}
+							}}
+						/>
+					</>
+				)}
+				<SubmitButton variant="primary" onClick={handleSubmit}>
```

Comment:
I realize every time this is enabled all of the time with the same text. Maybe we can add something to disable it between steps like

```ts
const submitEnabled = (email.value != "" && !email.error && !codeStep) || (codeStep && validationCode.value != "" && !validationCode.error) || (passStep && password.value != "" && !password.error && password.value == confirmPass.value)
```
This should work to enable only when the fields are filled out for the appropriate step. You could add something like this to ~ line 107 ish and then just add the following:

```suggestion
				<SubmitButton variant="primary" onClick={handleSubmit} disabled={!submitEnabled}>
```